tests/formatter.rs

```diff
@@ -439,6 +439,37 @@ error:
     assert_data_eq!(renderer.render(snippets), expected);
 }
 
+#[test]
+fn annotate_newline_empty_span() {
+    let message = &[Group::with_title(Level::ERROR.title("bad")).element(
+        Snippet::source("\n\n\n\n\n\n\n")
+            .path("test.txt")
+            .annotation(AnnotationKind::Primary.span(0..0)),
+    )];
+
+    let expected_ascii = str![[r#"
+error: bad
+ --> test.txt:1:1
+  |
+1 |
+  | ^
+"#]];
+
+    let renderer = Renderer::plain();
+    assert_data_eq!(renderer.render(message), expected_ascii);
+
+    let expected_unicode = str![[r#"
+error: bad
+  ╭▸ test.txt:1:1
+  │
+1 │
+  ╰╴━
```

Comment:
Nothing gets underlined in unicode?

Reply:
I am not sure what you mean, the [underline primary character](https://github.com/rust-lang/annotate-snippets-rs/blob/66e67f506b3c1bb98f2853b97c4f3509e38e1227/src/renderer/mod.rs#L2570) (`━`) is there.